src/shape.cpp

```diff
@@ -95,7 +95,9 @@ struct shape_impl
         }
     }
 
-    shape_impl(const std::vector<shape>& subs) : m_type(shape::tuple_type), m_shapes(subs) {}
+    explicit shape_impl(const std::vector<shape>& subs) : m_type(shape::tuple_type), m_shapes(subs)
```

Comment:
This shouldn't need explicit here, this is just a private class.

Reply:
Sure I'll change.

Reply:
Done